File: 코딩 기초 트레이닝/등차수열의 특정한 항만 더하기-p181931/등차수열의 특정한 항만 더하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181931

def solution(a, d, included):
    current = 0
    answer = 0

    for index, value in enumerate(included):
        if index == 0:
            current += a
        else:
            current += d

        answer += current if value else 0


    return answer

File: 코딩 기초 트레이닝/등차수열의 특정한 항만 더하기-p181931/등차수열의 특정한 항만 더하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181931

function solution(a, d, included) {
    let answer = 0;
    let current = 0;

    included.forEach((value, index) => {
        current += index === 0 ? a : d;
        answer += value ? current : 0;
    });

    return answer;
}
